File: java/dataStructures/stack/Stack.java
class Stack {
    int n;
    int[] arr;
    int top;

    Stack(int n) {
        this.n = n;
        this.arr = new int[n];
        this.top = -1;
    }

    public void push(int pushedElement) {
        if (top >= n - 1) {
            System.out.println("Stack is full");
            return;
        }
        arr[++top] = pushedElement;
    }

    public int pop() {
        if (top < 0) {
            System.out.println("Stack is empty");
            return -1;
        }
        return arr[top--];
    }

    public int peek() {
        if (top < 0) {
            System.out.println("Stack is Empty");
            return -1;
        }
        return arr[top];
    }

    public static void main(String[] args) {
        Stack stack = new Stack(5);

        stack.push(1);
        stack.push(2);
        stack.push(3);

        System.out.println(stack.pop());
        stack.push(4);
        System.out.println(stack.pop());
        System.out.println(stack.pop());
        System.out.println(stack.pop());
    }
}

File: python/dataStructures/stack/stack.py
class Stack:
    class Node:
        def __init__(self, data):
            self.data = data
            self.next = None
    
    def __init__(self):
        self.head = None
    
    def push(self, data):
        new_node = self.Node(data)
        new_node.next = self.head
        self.head = new_node 
    
    def pop(self):
        if self.head is None:
            print('Stack is already empty')
            return
        value = self.head.data
        self.head = self.head.next
        return value

    def print_stack(self):
        if self.head is None:
            print('Stack is empty')
            return
        current_node = self.head
        while current_node is not None:
            print(f"{current_node.data} -> ", end="")
            current_node = current_node.next

stack = Stack()
stack.push(1)
stack.push(2)
stack.push(3)
stack.push(4)

print(stack.print_stack())
print(stack.pop())
print(stack.print_stack())